Autopilot/PathManager/Inc/waypointManager.hpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * Waypoint Manager Methods and Helpers
+ * Author: Dhruv Rawat
+ * Created: November 2020
+ * Last Updaetd: November 2020
+ */
+
+#ifndef WAYPOINT_MANAGER_H
+#define WAYPOINT_MANAGER_H
+
+#include <stdlib.h>
+
+#define ORBIT_FOLLOWING 0
+#define LINE_FOLLOWING 1
+
+#define PI 3.14159265359
+#define MAX_PATH_APPROACH_ANGLE PI/2
+
+#define PATH_BUFFER_SIZE 100
+
+//Waypoint Types
+#define DEFAULT_WAYPOINT 0 // Used to navigate flight path
+#define WAYPOINT_UNUSED 1
```

Comment:
Where can the user make use of these ?

Reply:
First off, I will change WAYPOINT_UNUSED to WAYPOINT_USED

These are used in the type parameter of the waypoint structure. When creating the object, if type = 1, the waypoint management will treat it like a regular waypoint. if type = 2, then the plane will circle when  it reaches the waypoint

Now, as the waypoint manager uses up waypoints, it'll set the type parameter to 1 just to signal that it has been used.